cg/clients/freshdesk/models.py

```diff
@@ -19,9 +18,27 @@ class TicketCreate(BaseModel):
     subject: str
     tags: list[str] = []
     type: str | None = None
+    custom_fields: Dict[str, Union[str, int, float, None]] = Field(default_factory=dict)
+
+    def to_multipart_data(self) -> List[Tuple[str, Union[str, int, bytes]]]:
+        """Convert the TicketCreate model into a list of tuples for multipart form data required by the Freshdesk API."""
+        multipart_data = []
+
+        for field, value in self.model_dump(exclude_none=True).items():
+            if isinstance(value, list) and field == "tags":
+                multipart_data.extend([("tags[]", tag) for tag in value])
+            elif isinstance(value, dict) and field == "custom_fields":
+                multipart_data.extend(
+                    [(f"custom_fields[{key}]", val) for key, val in value.items()]
+                )
+            else:
+                multipart_data.append((field, value))
+
+        return multipart_data
```

Comment:
We could consider using a [custom serialiser](https://docs.pydantic.dev/latest/api/functional_serializers/#pydantic.functional_serializers.model_serializer) to make model_dump() create multipart forms